internal/ptx/snowflake.go

```diff
@@ -2,20 +2,130 @@ package ptx
 )
 
-// SnowflakeDialer is a dialer for snowflake. When optional fields are
-// not specified, we use defaults from the snowflake repository.
+// SnowflakeRendezvousMethod is the method which with we perform the rendezvous.
+type SnowflakeRendezvousMethod interface {
+	// Name is the name of the method.
+	Name() string
+
+	// AMPCacheURL returns a suitable AMP cache URL.
+	AMPCacheURL() string
+
+	// BrokerURL returns a suitable broker URL.
+	BrokerURL() string
+
+	// FrontDomain returns a suitable front domain.
+	FrontDomain() string
+}
+
+// NewSnowflakeRendezvousMethodDomainFronting is a rendezvous method
+// that uses domain fronting to perform the rendezvous.
+func NewSnowflakeRendezvousMethodDomainFronting() SnowflakeRendezvousMethod {
+	return &snowflakeRendezvousMethodDomainFronting{}
+}
+
+type snowflakeRendezvousMethodDomainFronting struct{}
+
+func (d *snowflakeRendezvousMethodDomainFronting) Name() string {
+	return "domain_fronting"
+}
+
+func (d *snowflakeRendezvousMethodDomainFronting) AMPCacheURL() string {
+	return ""
+}
+
+func (d *snowflakeRendezvousMethodDomainFronting) BrokerURL() string {
+	return "https://snowflake-broker.torproject.net.global.prod.fastly.net/"
+}
+
+func (d *snowflakeRendezvousMethodDomainFronting) FrontDomain() string {
+	return "cdn.sstatic.net"
+}
+
+// NewSnowflakeRendezvousMethodAMP is a rendezvous method that
+// uses the AMP cache to perform the rendezvous.
+func NewSnowflakeRendezvousMethodAMP() SnowflakeRendezvousMethod {
+	return &snowflakeRendezvousMethodAMP{}
+}
+
+type snowflakeRendezvousMethodAMP struct{}
+
+func (d *snowflakeRendezvousMethodAMP) Name() string {
+	return "amp"
+}
+
+func (d *snowflakeRendezvousMethodAMP) AMPCacheURL() string {
+	return "https://cdn.ampproject.org/"
+}
+
+func (d *snowflakeRendezvousMethodAMP) BrokerURL() string {
+	return "https://snowflake-broker.torproject.net/"
+}
+
+func (d *snowflakeRendezvousMethodAMP) FrontDomain() string {
+	return "www.google.com"
+}
+
+// ErrSnowflakeNoSuchRendezvousMethod indicates the given rendezvous
+// method is not supported by this implementation.
+var ErrSnowflakeNoSuchRendezvousMethod = errors.New("ptx: unsupported rendezvous method")
+
+// NewSnowflakeRendezvousMethod creates a new rendezvous method by name. We currently
+// support the following rendezvous methods:
+//
+// 1. "domain_fronting" uses domain fronting with the sstatic.net CDN;
+//
+// 2. "" means default and it is currently equivalent to "domain_fronting" (but
+// we don't guarantee that this default may change over time);
+//
+// 3. "amp" uses the AMP cache.
+//
+// Returns either a valid rendezvous method or an error.
+func NewSnowflakeRendezvousMethod(method string) (SnowflakeRendezvousMethod, error) {
+	switch method {
+	case "domain_fronting", "":
+		return NewSnowflakeRendezvousMethodDomainFronting(), nil
+	case "amp":
+		return NewSnowflakeRendezvousMethodAMP(), nil
+	default:
+		return nil, ErrSnowflakeNoSuchRendezvousMethod
+	}
+}
```

Comment:
All of this is new code that needs to be tested. I chose to make the rendezvous strategy configurable because it seems to me we want to leverage this opportunity based on the convo at https://github.com/ooni/probe/issues/2004.

I am also wondering whether we want (maybe at a later stage?) to try different strategies each time we run torsf rather than just defaulting to domain_fronting + persistent data directory.